conflation/aggregation.py

```diff
@@ -0,0 +1,139 @@
+import copy
+import json
+import os
+import pandas as pd
+import pickle
+
+from mapillary import util
+
+MAP_MATCH_COLS = ["density", "road_class", "type", "kph"]
+# Basic config from OpenStreetMapSpeeds/schema repo where all values are None.
+BASE_CONFIG = {
+    "rural": {
+        "way": [None, None, None, None, None, None, None, None],
+        "link_exiting": [None, None, None, None, None],
+        "link_turning": [None, None, None, None, None],
+        "roundabout": [None, None, None, None, None, None, None, None],
+        "driveway": None,
+        "alley": None,
+        "parking_aisle": None,
+        "drive-through": None,
+    },
+    "suburban": {
+        "way": [None, None, None, None, None, None, None, None],
+        "link_exiting": [None, None, None, None, None],
+        "link_turning": [None, None, None, None, None],
+        "roundabout": [None, None, None, None, None, None, None, None],
+        "driveway": None,
+        "alley": None,
+        "parking_aisle": None,
+        "drive-through": None,
+    },
+    "urban": {
+        "way": [None, None, None, None, None, None, None, None],
+        "link_exiting": [None, None, None, None, None],
+        "link_turning": [None, None, None, None, None],
+        "roundabout": [None, None, None, None, None, None, None, None],
+        "driveway": None,
+        "alley": None,
+        "parking_aisle": None,
+        "drive-through": None,
+    },
+}
+ROAD_CLASS_INDEX_MAPPING = {
+    "motorway": 0,
+    "trunk": 1,
+    "primary": 2,
+    "secondary": 3,
+    "tertiary": 4,
+    "unclassified": 5,
+    "residential": 6,
+    "service_other": 7,
+}
+
+
+def run(map_matches_dir: str, results_dir: str) -> None:
```

Comment:
the config format supports varying levels of granularity. specifically im refering to this bit from the readme:

> These geographies are treated as a hierarchy to allow varying degrees of specificity. For example one can specify a global set of speeds (i.e. applies to the whole world) by simply omitting both the iso3166-1 and iso3166-2 keys in a given object. To specify speeds across an entire country one needs to only populate the iso3166-1 key and omit the iso3166-2 key. For full granularity one must include both iso3166-1 and iso3166-2.

At the moment this only creates data for the most specific, ie both country and primary subdivision are present. We should keep counts of each item in these specific configs so we can then generate from them the country only configs as well as the fall back global config (ie no country and no state). By keeping the counts of each metric we can easily do a weighted average.

I dont think we need to solve it in this PR but if we dont do it, there will be a pretty major degradation in quality mainly because we cant guarantee that the admin information is stable in someone else's implementation on osm data. Maybe someone only has country level data in their downstream project, or maybe their admin data is missing the principal subdivision information in france or they use a different subdivision all together with different names. whatever the case may be its important to provide a reasonable fall back. if you look at the `iso3166-2` information that valhalla gets from osm boundary information you'll see its far from "clean".

Reply:
Added country-level aggregation since it was fairly straightforward with the new `aggregation.py` implementation. Added a TODO for overall, we can probably create a new pandas DF with each country's results + a count for weight and then at the end do a quick weighted average.